src/Microsoft.Health.Fhir.SpecManager/Language/TypeScript.cs

```diff
@@ -475,25 +475,30 @@ private void WriteExpandedResourceInterfaceBinding()
                 exportName = complex.NameForExport(FhirTypeBase.NamingConvention.PascalCase, true);
                 string typeName = complex.TypeForExport(FhirTypeBase.NamingConvention.PascalCase, _primitiveTypeMap, false);
 
-                _writer.WriteLineIndented($"export interface {exportName} extends {typeName} {{");
+                if (ShouldSupportGenerics(exportName)) {
+                    _writer.WriteLineIndented($"export interface {exportName}<T extends {typeName} = Resource> {{");
```

Comment:
I still believe that the generic default should be `FhirResource`.
Initially, @bkaney suggested [using Resource](https://github.com/Automate-Medical/fhir-codegen/commit/8f61e67eb23a814d84546180420c7c97f76b263a#r52356797)

But he refers to the type of BundleEntry itself, we are not using this generic for BundleEntry.
We use it for [BundleEntry.resource](https://www.hl7.org/fhir/bundle-definitions.html#Bundle.entry.resource) that has [Resource type from FHIR spec](https://www.hl7.org/fhir/resource.html). **Resource type from FHIR spec** is represented as **FhirResource** type in `@types/fhir`.